### PR TITLE
Adds Hypernoblium interactions with Plasmamans to stop self combustion.

### PR DESCRIPTION
**Pull request by GuillaumePrata**
## About The Pull Request
Plasmamen won't interact with O2 if there is also Hypernoblium in the air and if already on fire, will get extinguished.
Adds Hypernoblium as a reagent when inhaled, having Hypernoblium in your system as a Plasmamen will also stop combustion.

## Why It's Good For The Game
Extra utility for Hypernoblium and possibly some fun niche things for Atmos to work toward, build a Hypernoblium bar to hang out, make Hypernoblium/Plasma internals so they can throw their suits away and enjoy the Holodeck Beach.
Flood the station with Hypernoblium to make a Plasmaman heaven!

Also stops the "Your body reacts with the atmosphere and bursts into flames!" spam even if you were fireproof.

## Changelog
:cl: Guillaume H. Prata
add: Hypernoblium now turns into a reagent when inhaled, which stops the combustion if on a Plasmaman system.
add: Plasmaman interaction with Hypernoblium in the air, stop combustion and extinguishes if they are already on fire.
/:cl:

### CODE REVIEW

File: code/modules/surgery/organs/lungs.dm [hunk 1]
```diff
@@ -371,6 +371,14 @@
 
 		handle_breath_temperature(breath, H)
 		breath.garbage_collect()
+
+	// Hyper-Nob
+		gas_breathed = breath_gases[/datum/gas/hypernoblium][MOLES]
+		if (gas_breathed > gas_stimulation_min)
+			var/existing = H.reagents.get_reagent_amount(/datum/gas/hypernoblium)
+			H.reagents.add_reagent(/datum/reagent/hypernoblium,max(0, 1 - existing))
+		breath_gases[/datum/gas/hypernoblium][MOLES]-=gas_breathed
```
Comment (LemonInTheDark) [inline code/modules/surgery/organs/lungs.dm:380]: Can't remember, but what's the proportion of breath_gasses to the gasses around you?

File: code/modules/surgery/organs/lungs.dm [hunk 2]
```diff
@@ -371,6 +371,14 @@
...
+
```
Comment (LemonInTheDark) [inline code/modules/surgery/organs/lungs.dm:381]: Can't recall, what's the proportion of breath_gases to the air around you?

File: code/modules/surgery/organs/lungs.dm (hunk 1 above)
Comment (GuillaumePrata) [reply code/modules/surgery/organs/lungs.dm:380]: Its copy from Stimulum which I assume would be a balanced number.

Comment (LemonInTheDark) [reply code/modules/surgery/organs/lungs.dm:380]: aight that works

**Comment by infoman-cor** [comment]
Ah yes time to make gas/reagents that stop Felinids ears from getting fucked by flashbangs, lizards from getting cold in space and moths from getting fucked by light. All this really seems to do is just remove an interesting function of a race while they're breathing a gas. And plus if you want to build a plasmeme safe bar and such you can just open a plasma tank in it. While I really like the idea of gasses and reagents uniquely reacting with races I feel something that just straight up removes the only interesting trait they have is not a good idea.

**Comment by Tharcoonvagh** [comment]
I mostly agree with the above comment however opening a can of plasma and taking your suit off will only result in the surroundings burning as well

**Comment by msgerbs** [comment]
> 
> 
> Ah yes time to make gas/reagents that stop Felinids ears from getting fucked by flashbangs, lizards from getting cold in space and moths from getting fucked by light. All this really seems to do is just remove an interesting function of a race while they're breathing a gas. And plus if you want to build a plasmeme safe bar and such you can just open a plasma tank in it. While I really like the idea of gasses and reagents uniquely reacting with races I feel something that just straight up removes the only interesting trait they have is not a good idea.

This seems like a silly complaint considering this is likely just to be a lategame/extended fun thing. It's hardly removing their trait to give them an additional option to deal with it by cooperating with other departments.

**Comment by GuillaumePrata** [comment]
> Ah yes time to make gas/reagents that stop Felinids ears from getting fucked by flashbangs, lizards from getting cold in space and moths from getting fucked by light. All this really seems to do is just remove an interesting function of a race while they're breathing a gas.

There are reagents that are a lot more available than Hypernoblium will be that fix all these issues in game already, and multiple other options in game to fix a race weakness, I dont see why Plasmaman should be excluded from progression in this side of the gameplay progression in a round.

> I feel something that just straight up removes the only interesting trait they have is not a good idea.

I dont think that is their only interesting trait or that having to keep a suit all the time is that essential to Plasmaman, there are some ways to be a suitless Plasmaman in game already.

**Comment by Fikou** [comment]
> Ah yes time to make gas/reagents that stop Felinids ears from getting fucked by flashbangs, lizards from getting cold in space and moths from getting fucked by light. 

but felinids or moths dont have any features like that and we already have temperature stabilizing chems 🥴

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 3]
```diff
@@ -1333,6 +1333,23 @@
...
+/datum/reagent/hypernoblium
+	name = "Hyper-Noblium"
+	description = "A suppresive gas that stops gas reactions on those who inhale it."
+	reagent_state = GAS
+	metabolization_rate = REAGENTS_METABOLISM * 0.5 // Because stimulum/nitryl/freon/hyper-nob are handled through gas breathing, metabolism must be lower for breathcode to keep up
+	color = "90560B"
+	taste_description = "searingly cold"
+
+/datum/reagent/hypernoblium/on_mob_metabolize(mob/living/L)
+	..()
```
Comment (Rohesie) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:1345]:
```suggestion
	. = ..()
```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 4]
```diff
@@ -1333,6 +1333,23 @@
...
+	if(isplasmaman(L))
+		ADD_TRAIT(L, TRAIT_NOFIRE, type)
+
+/datum/reagent/hypernoblium/on_mob_end_metabolize(mob/living/L)
+	if(isplasmaman(L))
+		REMOVE_TRAIT(L, TRAIT_NOFIRE, type)
+	..()
```
Comment (Rohesie) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:1352]:
```suggestion
	return ..()
```

**Pull request merged**